app/services/foreman_templates/template_exporter.rb

```diff
@@ -107,5 +107,41 @@ def export_method
           raise "Unknown metadata export mode #{@metadata_export_mode}"
       end
     end
+
+    private
+
+    def find_templates
+      @organizations.empty? && @locations.empty? ? Template.all : find_taxed_templates
+    end
+
+    def find_taxed_templates
+      Template.subclasses.reduce([]) do |memo, subclass|
+        memo.concat templates_in_tax(subclass)
+      end
+    end
+
+    def templates_in_tax(template_class)
+      location_ids = taxes_ids('location')
+      organization_ids = taxes_ids('organization')
+      if location_ids.empty?
+        templates_query(template_class, organization_ids)
+      elsif organization_ids.empty?
+        templates_query(template_class, location_ids)
+      else
+        templates_query(template_class, organization_ids) & templates_query(template_class, location_ids)
+      end
+    end
+
+    def templates_query(template_class, tax_ids)
+      template_class.joins(:taxable_taxonomies).where(:taxable_taxonomies => { :taxonomy_id => tax_ids })
+    end
+
+    def taxes_ids(tax_type)
+      tax_type.capitalize.constantize
+              .where(:name => instance_variable_get("@#{tax_type.pluralize}")["#{tax_type}_names".to_sym])
```

Comment:
please define attr_accessor to avoid using instance_variable_get, public_send is more readable or even better would be something like `@parameters[tax_type]['names']` that would be created in action initializer, to avoid this dynamic programming and be less dependent on parameters structure.